docs: Add docs for setting store loader (#9526)
cosmos/cosmos-sdk: docs/core/upgrade.md

You can introduce entirely new modules to the application during an upgrade. New modules are recognized because they have not yet been registered in `x/upgrade`'s `VersionMap` store. In this case, `RunMigrations` calls the `InitGenesis` function from the corresponding module to set up its initial state.

### Add StoreUpgrades for New Modules

All chains preparing to run in-place store migrations will need to manually add store upgrades for new modules and then configure the store loader to apply those upgrades. This ensures that the new module's stores are added to the multistore before the migrations begin.

```golang
upgradeInfo, err := app.UpgradeKeeper.ReadUpgradeInfoFromDisk()
if err != nil {
	panic(err)
}

if upgradeInfo.Name == "my-plan" && !app.UpgradeKeeper.IsSkipHeight(upgradeInfo.Height) {
	storeUpgrades := storetypes.StoreUpgrades{
		// add store upgrades for new modules
		// Example: 

> 	  //    Added: []string{"foo", "bar"},

Author: ```suggestion
		//    Added: []string{"foo", "bar"},
```

			// ...
		}

		// configure store loader that checks if version == upgradeHeight and applies store upgrades
		app.SetStoreLoader(upgradetypes.UpgradeStoreLoader(upgradeInfo.Height, &storeUpgrades))
	}

> ```

Reviewer: nit: formatting is off.

You can introduce entirely new modules to the application during an upgrade. New modules are recognized because they have not yet been registered in `x/upgrade`'s `VersionMap` store. In this case, `RunMigrations` calls the `InitGenesis` function from the corresponding module to set up its initial state.

### Add StoreUpgrades for New Modules

> All chains which will run in-place store migrations will need to manually add storeloader for upgrades introducing new modules. So that the new module's stores are added to the multistore before the migrations begin.

Reviewer 2: ```suggestion
All chains preparing to run in-place store migrations will need to manually add store upgrades for new modules and then configure the store loader to apply those upgrades. This ensures that the new module's stores are added to the multistore before the migrations begin.
```

```golang

> upgradeInfo, err := app.UpgradeKeeper.ReadUpgradeInfoFromDisk()

Reviewer 2: ```suggestion
	upgradeInfo, err := app.UpgradeKeeper.ReadUpgradeInfoFromDisk()
```

	if err != nil {
		panic(err)
	}

	if upgradeInfo.Name == "my-plan" && !app.UpgradeKeeper.IsSkipHeight(upgradeInfo.Height) {
		storeUpgrades := storetypes.StoreUpgrades{

>       // ...
>       // add store upgrades for new modules
>       // Example: 
>       //    Added: []string{"foo"},
>       // ...

Reviewer 2: ```suggestion
		// ...
		// add store upgrades for new modules
		// Example: 
		//    Added: []string{"foo", "bar"},
		// ...
```

> 
>         // ...
>         // add store upgrades for new modules
>         // Example: 
>         //    Added: []string{"foo", "bar"},
>         // ...
> 

Reviewer 3: ```suggestion
			// add store upgrades for new modules
			// Example: 
			//    Added: []string{"foo", "bar"},
			// ...
```

Author: I don't know why the comments are appearing this way here, but they are in good indentation if you checkout to this branch and see them
